### DISCUSSION TITLE
TRPC's useContext does not invalidate query.

### DISCUSSION THREAD

**Discussion by stephenasuncionDEV** [trpc/trpc#4780] [Q&A] [answered]
Still dont know why `useContext` is not working for me (using .refetch, .invalidate on utils does nothing):

I can confirm that onSuccess gets called. It also doesnt work even if I remove the enabled option. It invalidates when I put an override on `createTRPCNext` but I really don't want to invalidate all the queries in the cache.
```ts
const utils = trpc.useContext();
const websites = trpc.website.getWebsites.useQuery(undefined, {
    enabled: isAuthenticated,
  });
 const updateWebsite = trpc.website.updateWebsite.useMutation({
    onSuccess: async () => {
      await utils.website.getWebsites.invalidate();
    },
  });
```

**Comment by stephenasuncionDEV** [comment, marked as answer]
Fixed: Another context was conflicting with TRPC because both use react-query.

**Reply by melodyclue** [reply]
hi, I have a same issue , what is  Another context ?

**Reply by stephenasuncionDEV** [reply]
Another library using Tanstack's query client which trpc also uses.

**Reply by AndrewMakeApp** [reply]
Could you share how you fixed it, we are having the same setup & issue

**Comment by BishalN** [comment]
Now this useContext seems to be deprecated how do you perform invalidation in trpc + nextjs 
other that using router.refresh ?

**Reply by stephenasuncionDEV** [reply]
useContext was renamed to `useUtils`
